mender-update/deployments.cpp

```diff
@@ -92,14 +92,17 @@ error::Error CheckNewDeployments(
 	if (provides.find("artifact_name") == provides.end()) {
 		return MakeError(InvalidDataError, "Missing artifact name data");
 	}
-	string artifact_name = provides["artifact_name"];
 
 	stringstream ss;
 	ss << R"({"update_control_map": false, "device_provides": {)";
 	ss << R"("device_type": ")";
 	ss << json::EscapeString(device_type);
-	ss << R"(", "artifact_name": ")";
-	ss << json::EscapeString(artifact_name);
+
+	for (const auto &kv : provides) {
+		ss << "\", \"" + json::EscapeString(kv.first) + "\": \"";
```

Comment:
Suggest sending it minified (no spaces) simply to save space.

Reply:
I actually had it that way initially, but it looked ugly in the tests. I'll happily change it back, though :+1: